Framework/Algorithms/src/CreateDetectorTable.cpp

```diff
@@ -317,6 +331,9 @@ void populateTable(ITableWorkspace_sptr &t, const MatrixWorkspace_sptr &ws, cons
       if (includeDiffConstants) {
         colValues << 0.0 << 0.0 << 0.0 << 0.0;
       }
+      if (includeDetectorPosition) {
+        colValues << V3D(0.0, 0.0, 0.0);
```

Comment:
I don't think the test you added will hit this branch will it? 
I've manually tested this using `PEARL00112777.nxs` (e.g. spectrum index 5) - so it works! But goof practice to try and cover every line you add with a test (not always easy though).